### ISSUE TITLE
Enter width of platforms in the format that is standard for OSM (comma-seperated meters)

### ISSUE THREAD

**Issue by wielandb** [OPENER-next/OpenStop#128]
### Problem

After answering questions about the width of a bus stop, I got a comment on the changeset telling me that "498 cm" is non-standard for OSM and should be avoided. Instead, "4.98" should be the the value entered into the Database. 

### Which app version do you use?

0.3.0+8

### What operating system do you use?

Android

### What operating system *version* do you use?

10

### Additional information

[Link to the CS in concern](https://www.openstreetmap.org/changeset/127891997)

**Comment by Robbendebiene** [comment]
After revisiting this I see two solutions:

## Solution 1:
Add an expression to convert the input values from cm to m.

**Expression implementation**:

```dart
String? _math(Iterable<String> args) {
  final iter = args.iterator..moveNext();
  try {
    var result = num.parse(iter.current);

    while(iter.moveNext()) {
      final operator = iter.current;
      iter.moveNext();
      final value = num.parse(iter.current);

      switch (operator) {
        case '+': result += value; break;
        case '-': result -= value; break;
        case '*': result *= value; break;
        case '/': result /= value; break;
        case '%': result %= value; break;
      }
    }
    // if result has no decimal places, convert and return as int to avoid empty 0 decimal
    return result == result.roundToDouble()
      ? result.toInt().toString()
      : result.toString();
  }
  catch(e) {
    debugPrint(e.toString());
    return null;
  }
}
```
This can be used in the question catalog (constructor) like this: `"width": ["math", "$input", "/", "100"]`

The problem is that calculating with doubles can always lead to undesired imprecisions. So we would need an additional package that handles computation without losing precision: https://pub.dev/packages/decimal
I also cannot think of any other use case for the already quite generalized "math" expression.

Another solution would be to make an expression that does no mathematical computation and solely adds a decimal point at a certain position. Could look something like this: `"width": ["insert/splice", "$input", "-2", "."]`

## Solution 2:
Keep everything as it is.

Even mm is a somehow an accepted value in OSM (https://wiki.openstreetmap.org/wiki/Category:Keys_indicating_lengths_in_millimetres) and suggested in other wiki pages: https://wiki.openstreetmap.org/wiki/Key:diameter#Unit_of_measure

I feel like no one would be surprised about seeing a cm unit.

```
[out:json][timeout:2665];
(
  nwr["width"~"cm$"];
  nwr["length"~"cm$"];
  nwr["height"~"cm$"];
);
// print results
out count;

```
-> At least 2284 elements with cm units (even though a good amount of them come from historic landmarks in Switzerland)

Maybe there is even a chance to add cm to the list of explicitly allowed units by starting a proposal or something similar.

**Comment by wielandb** [comment]
Did #183 make this issue closable?

**Comment by Robbendebiene** [comment]
Resolved by #183
